File: src/main/java/org/example/eventhub/entity/Location.java
package org.example.eventhub.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Embeddable
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Location {

    @Column(length = 100)
    @NotBlank
    private String city;

    @Column(length = 200)
    @NotBlank
    private String street;

    @Column(length = 20)
    @NotBlank
    private String house;

    private String additionalInfo;
}

File: src/main/java/org/example/eventhub/entity/Location.kt
package org.example.eventhub.entity

import jakarta.persistence.Column
import jakarta.persistence.Embeddable

@Embeddable
class Location(
    @Column(length = 100)
    var city: String? = null,

    @Column(length = 200)
    var street: String? = null,

    @Column(length = 20)
    var house: String? = null,

    var additionalInfo: String? = null
)